[aggregator] Add python bindings for service checks submission

olivielpeau opened the pull request:
### What does this PR do?

* Adds python bindings for service checks submission
* Fixes an issue when the passed tags list is `None`

### Testing Guidelines

There are no automated tests, I did some manual testing. I could work on adding a basic test though (loading a py check that uses the `service_check` method and running it ; that would check that the submission doesn't crash the agent at least).

### Additional Notes

I've added a check on whether `tags` is `None`, I'm not sure it's the best way to do it though.

olivielpeau committed:
[aggregator] Add python bindings for service checks submission

olivielpeau committed:
[core][py] Check that `tags` list is not `None` before getting its size

truthbk reviewed (commented):
Just a minor note regarding the `bool` type.

masci reviewed (changes requested):
A small comment to support the build on Windows othewise 👍

olivielpeau committed:
[core][py] Remove C99 flag

And remove the related `stdbool` usage, it was the only C99 feature
we were using

olivielpeau committed:
[core][py][windows] Add cgo preamble

Fixes the Mingw failures at link time

masci reviewed (approved):
:shipit:
